core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/logprovider/provider.go

```diff
@@ -196,7 +195,7 @@ func (p *logEventProvider) ReadLogs(pctx context.Context, ids ...*big.Int) error
 	ctx, cancel := context.WithTimeout(pctx, readLogsTimeout)
 	defer cancel()
 
-	latest, err := p.poller.LatestBlock(pg.WithParentCtx(ctx))
+	latest, err := p.poller.LatestBlock(pctx)
```

Comment:
Wouldn't we want to use `ctx` here too?
```suggestion
	latest, err := p.poller.LatestBlock(ctx)
```

Reply:
Nice catch!